sdks/python/apache_beam/io/fileio.py

```diff
@@ -95,18 +87,15 @@ def is_valid_compression_type(cls, compression_type):
   @classmethod
   def mime_type(cls, compression_type, default='application/octet-stream'):
     mime_types_by_compression_type = {
-        cls.GZIP: 'application/x-gzip',
-        cls.ZLIB: 'application/octet-stream',
         cls.BZIP2: 'application/x-bz2',
+        cls.GZIP: 'application/x-gzip',
     }
     return mime_types_by_compression_type.get(compression_type, default)
 
   @classmethod
   def detect_compression_type(cls, file_path):
     """Returns the compression type of a file (based on its suffix)"""
-    compression_types_by_suffix = {'.gz': cls.GZIP,
-                                   '.z': cls.ZLIB,
```

Comment:
Yes, it is intentional.

We are only supporting the GZIP standard (which is more or less the same compression as ZLIB, just different headers/checksup) but also more widely compatible with external tools.

This also makes Dataflow Java and Dataflow Python in-sync with each other as well as with the Dataflow service which only recognized BZIP2 and GZIP as compressed text files at the moment.